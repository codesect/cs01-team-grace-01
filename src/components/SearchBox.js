import React, { useState } from 'react';

import PropTypes from 'prop-types';

export const SearchBox = props => {
  const [query, setQuery] = useState('');
  return (
    <form
      onSubmit={e => {
        e.preventDefault();
        props.onSubmit(query);
        setQuery('');
      }}
    >
      <label>
        Start by searching for a phone number
        <input
          type="text"
          placeholder="Type a phone number"
          onChange={e => setQuery(e.target.value)}
          value={query}
        />
      </label>
      <button type="submit">Search</button>
    </form>
  );
};

SearchBox.prototypes = {
  onSubmit: PropTypes.func.isRequired,
};
