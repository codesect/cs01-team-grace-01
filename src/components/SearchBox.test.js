import React from 'react';
import { shallow, mount } from 'enzyme';
import { SearchBox } from './SearchBox';

describe('<SearchBox />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<SearchBox />);

    expect(wrapper).toMatchSnapshot();
  });

  it('updates state on input change and calls onSubmit prop when form is submitted', () => {
    const handleSubmit = jest.fn();
    const phoneNumber = '03446405416';
    const wrapper = mount(<SearchBox onSubmit={handleSubmit} />);
    const form = wrapper.find('form');
    form.find('input').simulate('change', { target: { value: phoneNumber } });

    form.simulate('submit');

    expect(handleSubmit).toHaveBeenCalledTimes(1);
    expect(handleSubmit).toBeCalledWith(phoneNumber);
  });

  it('render input field with a corresponding label', () => {
    const wrapper = shallow(<SearchBox />);

    const label = wrapper.find('label');
    expect(label).toHaveLength(1);

    const input = label.find('input[type="search"]');
    expect(input).toHaveLength(1);
  });
});
