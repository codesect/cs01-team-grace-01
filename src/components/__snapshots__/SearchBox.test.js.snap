// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Start by searching for a phone number
    <input
      onChange={[Function]}
      placeholder="Type a phone number"
      type="search"
      value=""
    />
  </label>
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
